runtime/lua/vim/lsp.lua

```diff
@@ -453,13 +464,14 @@ local function on_client_exit(code, signal, client_id)
   end)
 end
 
---- Starts and initializes a client with the given configuration.
+--- @private
+--- Creates and initializes a client with the given configuration.
 --- @param config vim.lsp.ClientConfig Configuration for the server.
 --- @return integer? client_id |vim.lsp.get_client_by_id()| Note: client may not be
 ---         fully initialized. Use `on_init` to do any actions once
 ---         the client has been initialized.
 --- @return string? # Error message, if any
-function lsp.start_client(config)
+function lsp._create_and_initialize_client(config)
```

Comment:
I think you could move this above lsp.start() as `local function start_client(config)`